homework/index.js

```diff
@@ -30,13 +33,94 @@
     return elem;
   }
 
+  // deletes the previous repository table if selected repository changes
+  function removeChildElements(parent) {
+    while (parent.firstChild) {
+      parent.removeChild(parent.firstChild);
+    }
+  }
+  // to show the information of selected repository in a table
+  function showRepoInfo(index, repoDiv, repoArray, containerDiv) {
+    removeChildElements(repoDiv);
+    const selectedRepo = repoArray[index];
+    const headerArray = ['Repository', 'Description', 'Forks', 'Updated'];
+    const repoInfo = [
+      selectedRepo.name,
+      selectedRepo.description,
+      selectedRepo.forks,
+      selectedRepo.updated_at,
+    ];
+    const table = createAndAppend('table', repoDiv);
+    for (let i = 0; i < headerArray.length; i++) {
+      const tr = createAndAppend('tr', table);
+      createAndAppend('td', tr, { text: headerArray[i], class: 'label' });
+      if (i === 0) {
+        const repoLink = createAndAppend('td', tr);
+        createAndAppend('a', repoLink, {
+          href: `${selectedRepo.svn_url}`,
+          target: '_blank',
+          text: repoInfo[i],
+        });
+      } else {
+        createAndAppend('td', tr, { text: repoInfo[i] });
+      }
+    }
+    // to fetch and display information about contributors
+    fetchJSON(selectedRepo.contributors_url, (err, data) => {
+      if (err) {
+        createAndAppend('div', containerDiv, { text: err.message, class: 'alert-error' });
+      } else {
+        const contributorsDiv = createAndAppend('div', containerDiv, { class: 'right_div' });
+
+        createAndAppend('p', contributorsDiv, {
+          text: 'Contributors',
+          class: 'contributor_header',
+        });
+
+        const str = JSON.stringify(data, null, 2);
```

Comment:
Why `stringify` the data only to `parse` it on the next line? The data and contributorsArray will look the same. Can you elaborate?

Reply:
Last week there was a misunderstood about stringify and parse, but now i got the point and corrected the code.